pkg/operator/certrotationcontroller/satokensigner_controller.go

```diff
@@ -188,7 +190,12 @@ func (c *SATokenSignerController) syncWorker() error {
 	if err != nil && !errors.IsNotFound(err) {
 		return err
 	}
-	needNewSATokenSigningKey := errors.IsNotFound(err) || len(saTokenSigner.Data["service-account.key"]) == 0 || len(saTokenSigner.Data["service-account.pub"]) == 0
+	needNewSATokenSigningKey := errors.IsNotFound(err)
+	matchingKeyPair, err := isValidKeyPair(saTokenSigner.Data["service-account.pub"], saTokenSigner.Data["service-account.key"])
```

Comment:
if key itself is not present, then a function call to  `isValidKeypair` can be skipped.